File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/details/DetailActivity.java
package com.jaynewstrom.concretesample.details;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.widget.ListView;

import com.jaynewstrom.concretesample.BaseActivity;
import com.jaynewstrom.concretesample.R;
import com.jaynewstrom.concretesample.application.ApplicationComponent;

import butterknife.BindView;
import butterknife.ButterKnife;

public final class DetailActivity extends BaseActivity<DetailComponent> {

    private static final String INTENT_KEY_TITLE = "intent.title";

    public static Intent buildIntent(Context context, String detailsTitle) {
        Intent intent = new Intent(context, DetailActivity.class);
        intent.putExtra(INTENT_KEY_TITLE, detailsTitle);
        return intent;
    }

    @BindView(R.id.list_view) ListView listView;

    @Override protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_details);
        ButterKnife.bind(this);
        setTitle(getIntent().getStringExtra(INTENT_KEY_TITLE));
        listView.setAdapter(new DetailsListAdapter());
    }

    @Override protected DetailConcreteBlock concreteBlock(ApplicationComponent applicationComponent) {
        return new DetailConcreteBlock(applicationComponent, getIntent().getStringExtra(INTENT_KEY_TITLE));
    }

    @Override protected void performInject(DetailComponent component) {
        component.inject(this);
    }
}

File: concrete-sample/src/main/java/com/jaynewstrom/concretesample/details/DetailActivity.kt
package com.jaynewstrom.concretesample.details

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.ListView
import com.jaynewstrom.concretesample.BaseActivity
import com.jaynewstrom.concretesample.R
import com.jaynewstrom.concretesample.application.ApplicationComponent

class DetailActivity : BaseActivity<DetailComponent>() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_details)
        title = intent.getStringExtra(INTENT_KEY_TITLE)
        val listView = findViewById<ListView>(R.id.list_view)
        listView.adapter = DetailsListAdapter()
    }

    override fun concreteBlock(applicationComponent: ApplicationComponent): DetailConcreteBlock {
        return DetailConcreteBlock(applicationComponent, intent.getStringExtra(INTENT_KEY_TITLE)!!)
    }

    override fun performInject(component: DetailComponent) {
        component.inject(this)
    }

    companion object {
        private const val INTENT_KEY_TITLE = "intent.title"

        fun buildIntent(context: Context, detailsTitle: String): Intent {
            val intent = Intent(context, DetailActivity::class.java)
            intent.putExtra(INTENT_KEY_TITLE, detailsTitle)
            return intent
        }
    }
}
